conda/core/solve.py

```diff
@@ -731,15 +739,18 @@ def _add_specs(self, ssc):
         # the only things we should consider freezing are things that don't conflict with the new
         #    specs being added.
         explicit_pool = ssc.r._get_package_pool(self.specs_to_add)
+        if ssc.prune:
+            # Ignore installed specs on prune.
+            installed_specs = ()
+        else:
+            installed_specs = tuple(
+                chain.from_iterable(
+                    _.to_match_spec() for _ in ssc.prefix_data.iter_records()
+                )
+            )
```

Comment:
Actually this is incorrect, `Record.to_match_spec()` doesn't return an iterable, it only returns a single spec, no chaining is necessary:

```suggestion
            installed_specs = [
                record.to_match_spec() for record in ssc.prefix_data.iter_records()
            ]
```

Reply:
Thanks for the re-naming, that helps clarifies what's going on 👍